docs/concept/flavors.soy

```diff
@@ -0,0 +1,458 @@
+correctness, but avoids codegen (which is the slowest phase of Rust 
+compilation). This flavor invokes check on the suffixed build target.
+</p>
+
+{literal} 
+<pre>
+save-analysis
+</pre>
+{/literal}
+<p>
+The <code>#save-analysis</code> flavor is an extension of #check. It performs 
+the same actions as a #check build, but also emits a .json file containing 
+full type/symbol cross-reference information, for consumption by tools like 
+RLS (Rust Language Services).
+</p>
+
+{literal}
+<pre>
+doc
+</pre>
+{/literal}
+
+<p>
+The <code>#doc</code> flavor is the equivalent of the cargo doc command. It 
+uses the rustdoc tool to generate documentation for a target and its dependencies.
+</p>
+
+<h3>Compilation database</h3>
+<p>
+If you specify one of these flavors, Buck generates analysis information about 
+your build.
+</p>
+{literal}
+<pre>
+compilation-database
+</pre>
+{/literal}
+
+<p>
+Produces a JSON file that contains (an approximation of) the compiler commands 
+for compiling each file. The output is in 
+the <a href="https://eli.thegreenplace.net/2014/05/21/compilation-databases-for-clang-based-tools">
+clang compilation database</a> format.
+</p>
+
+<h3>Infer</h3>
+{literal}
+<pre>
+infer-analyze
+infer-capture-all
+</pre>
+{/literal}
+
+<p>
+These flavors run <a href="https://fbinfer.com">Facebook's Infer tool</a> and 
+generate intermediate Infer output.
+</p>
+
+<h2 id="symbol_http">Symbol stripping flavors<a class="inline-link" href="#symbol_http">#</a></h2>
+<p>
+These flavor control how symbols are stripped from an output binary.
+</p>
+
+<h3>Strip debug</h3>
+{literal}
+<pre>
+strip-debug
+</pre>
+{/literal}
+
+<p>
+Strip debug symbols; equivalent to <code>strip -S</code>.
+</p>
+
+<h3>Strip debug and non-external</h3>
+{literal} 
+<pre>
+strip-non-global
+</pre>
+{/literal}
+
+<p>
+Strip debug and non-external symbols; equivalent to <code>strip -x</code>.
+</p>
+
+<h3>Strip all</h3>
+{literal} 
+<pre>
+strip-all
+</pre>
+{/literal}
+<p>
+Strip all the things; equivalent to strip with no arguments.
+</p>
+
+<h2 id="header_http">Header Flavors<a class="inline-link" href="#header_http">#</a></h2>
+<p>
+These flavors are used with C++ header files, in the deps or exported_deps 
+attributes of a Buck target.
+</p>
+
+<h3>Headers used to compile a libraries own files</h3>
+{literal}
+<pre>
+private-headers
+</pre>
+{/literal}
+<p>
+Symlink tree, or header map, of headers used for compiling a library's own files.
+</p>
+
+<h3>Headers used to compile files that depend on this library</h3>
+{literal} 
+<pre>
+headers
+</pre>
+{/literal}
+
+<p>
+Symlink tree, or header map, of headers used to compile files for other 
+libraries that depend on this library.
+</p>
+
+<h2 id="apple_debug_http">Apple debug flavors<a class="inline-link" href="#apple_debug_http">#</a></h2>
+<p>
+Selects the actual binary used to represent a {call buck.cxx_binary /}. These are used internally.
```

Comment:
"used internally"? Internally to Buck or internal at FB?

Reply:
Buck